packages/context/test/unit/resolver.test.ts

```diff
@@ -0,0 +1,181 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context} from '../../src/context';
+import {inject} from '../../src/inject';
+import {instantiateClass} from '../../src/resolver';
+
+describe('constructor injection', () => {
+  var ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+  });
+
+  it('can resolve constructor arguments', () => {
+    class TestClass {
+      constructor(@inject('foo') public foo: string) {
+      }
+    }
+
+    let t = instantiateClass<TestClass>(TestClass, ctx) as TestClass;
```

Comment:
Please note the return type of `instantiateClass` is `T | Promise<T>`. We have to cast it.